Compiling Machida fails

I am getting the below error message when i try to compile machinda. I  am trying to set up Wallaroo in my Linux box. Since i was not able to reopen the previous(#1876) issue i am creating a new issue.
```
Stack dump:
0.      Running pass 'Function Pass Manager' on module 'machida'.
1.      Running pass 'Loop Vectorization' on function '@Array_t2_U128_val_U128_val_Serialise'
Segmentation fault (core dumped)
../machida/Makefile:68: recipe for target '/home/ubuntu/wallaroo-tutorial/wallaroo/machida/build/machida' failed
make: *** [/home/ubuntu/wallaroo-tutorial/wallaroo/machida/build/machida] Error 1
```

I am using a Linux AWS box (Instance Type: m5.large)
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.3 LTS
Release:        16.04
Codename:       xenial

Terminal Output
```
ubuntu@ip-:~/wallaroo-tutorial/wallaroo/machida$ make
From https://github.com/WallarooLabs/pony-kafka
 * branch            HEAD       -> FETCH_HEAD
Updating f738312..b26d8ae
Fast-forward
 examples/performance/main.pony                              |  460 ++++++++++++++++++++++++++++++++++
 examples/simple/main.pony                                   |   13 +-
 misc/kafka/cleanup_data_files.sh                            |    9 +
 misc/kafka/create_replicate_topic.sh                        |   12 +
 misc/kafka/download_kafka_java.sh                           |   16 ++
 misc/kafka/kafka-server-0.properties                        |  125 ++++++++++
 misc/kafka/kafka-server-1.properties                        |  125 ++++++++++
 misc/kafka/kafka-server-2.properties                        |  125 ++++++++++
 misc/kafka/start_kafka_0.sh                                 |    9 +
 misc/kafka/start_kafka_1.sh                                 |    9 +
 misc/kafka/start_kafka_2.sh                                 |    9 +
 misc/kafka/start_zookeeper.sh                               |    8 +
 misc/kafka/zookeeper.properties                             |   20 ++
 pony-kafka/_test.pony                                       |  118 +++++----
 pony-kafka/compression/lz4.pony                             |   12 +-
 pony-kafka/compression/snappy.pony                          |   24 +-
 pony-kafka/compression/zlib.pony                            |   16 +-
 pony-kafka/custombuffered/_test.pony                        |   93 ++++++-
 pony-kafka/custombuffered/codecs/big_endian_decoder.pony    |  172 +++++++------
 pony-kafka/custombuffered/codecs/little_endian_decoder.pony |  188 +++++++-------
 pony-kafka/custombuffered/codecs/varint_decoder.pony        |   22 +-
 pony-kafka/custombuffered/iso_reader.pony                   |  226 +++++++++++++++++
 pony-kafka/custombuffered/reader.pony                       |  487 +-----------------------------------
 pony-kafka/custombuffered/val_reader.pony                   |  471 +++++++++++++++++++++++++++++++++++
 pony-kafka/custombuffered/writer.pony                       |   31 +--
 pony-kafka/customlogger/logger.pony                         |   10 +-
 pony-kafka/customnet/custom_tcp_connection_handler.pony     |   23 +-
 pony-kafka/customnet/custom_tcp_connection_notify.pony      |    8 +
 pony-kafka/customnet/mock_tcp_connection_handler.pony       |   21 ++
 pony-kafka/customnet/tcp_connection_handler.pony            |   18 ++
 pony-kafka/kafka_api.pony                                   |  653 ++++++++++++++++++++++++++++---------------------
 pony-kafka/kafka_broker_connection.pony                     |   62 +++--
 pony-kafka/kafka_config.pony                                |  475 +++++++++++++++++++++++++----------
 pony-kafka/kafka_config_cli_parser.pony                     |  395 ++++++++++++++++++++++++++++++
 pony-kafka/kafka_errors.pony                                |  371 ++++++++++++++++++++++------
 pony-kafka/kafka_handler.pony                               | 1656 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++----------------------------------------
 36 files changed, 4698 insertions(+), 1794 deletions(-)
 create mode 100644 examples/performance/main.pony
 create mode 100755 misc/kafka/cleanup_data_files.sh
 create mode 100755 misc/kafka/create_replicate_topic.sh
 create mode 100755 misc/kafka/download_kafka_java.sh
 create mode 100644 misc/kafka/kafka-server-0.properties
 create mode 100644 misc/kafka/kafka-server-1.properties
 create mode 100644 misc/kafka/kafka-server-2.properties
 create mode 100755 misc/kafka/start_kafka_0.sh
 create mode 100755 misc/kafka/start_kafka_1.sh
 create mode 100755 misc/kafka/start_kafka_2.sh
 create mode 100755 misc/kafka/start_zookeeper.sh
 create mode 100644 misc/kafka/zookeeper.properties
 create mode 100644 pony-kafka/custombuffered/iso_reader.pony
 create mode 100644 pony-kafka/custombuffered/val_reader.pony
 create mode 100644 pony-kafka/kafka_config_cli_parser.pony
Previous HEAD position was b26d8ae... Fix invalid throttling edge case
HEAD is now at f738312... Compatiblity with ponyc master and exhaustive match changes.
Building builtin -> /usr/lib/pony/0.21.0-4301.acd811b/packages/builtin
Building . -> /home/ubuntu/wallaroo-tutorial/wallaroo/machida
Building signals -> /usr/lib/pony/0.21.0-4301.acd811b/packages/signals
Building ponytest -> /usr/lib/pony/0.21.0-4301.acd811b/packages/ponytest
Building time -> /usr/lib/pony/0.21.0-4301.acd811b/packages/time
Building collections -> /usr/lib/pony/0.21.0-4301.acd811b/packages/collections
Building wallaroo_labs/mort -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo_labs/mort
Building wallaroo_labs/options -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo_labs/options
Building wallaroo -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo
Building wallaroo/ent/spike -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/ent/spike
Building wallaroo/ent/network -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/ent/network
Building wallaroo/core/common -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/core/common
Building wallaroo/core/boundary -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/core/boundary
Building buffered -> /usr/lib/pony/0.21.0-4301.acd811b/packages/buffered
Building net -> /usr/lib/pony/0.21.0-4301.acd811b/packages/net
Building wallaroo_labs/bytes -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo_labs/bytes
Building promises -> /usr/lib/pony/0.21.0-4301.acd811b/packages/promises
Building wallaroo_labs/time -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo_labs/time
Building wallaroo/ent/watermarking -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/ent/watermarking
Building wallaroo/core/invariant -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/core/invariant
Building wallaroo/core/routing -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/core/routing
Building wallaroo_labs/guid -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo_labs/guid
Building random -> /usr/lib/pony/0.21.0-4301.acd811b/packages/random
Building wallaroo/core/messages -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/core/messages
Building serialise -> /usr/lib/pony/0.21.0-4301.acd811b/packages/serialise
Building wallaroo/ent/data_receiver -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/ent/data_receiver
Building wallaroo/core/data_channel -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/core/data_channel
Building files -> /usr/lib/pony/0.21.0-4301.acd811b/packages/files
Building capsicum -> /usr/lib/pony/0.21.0-4301.acd811b/packages/capsicum
Building wallaroo/ent/recovery -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/ent/recovery
Building wallaroo/core/topology -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/core/topology
Building assert -> /usr/lib/pony/0.21.0-4301.acd811b/packages/assert
Building wallaroo/ent/rebalancing -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/ent/rebalancing
Building wallaroo/core/initialization -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/core/initialization
Building wallaroo/ent/cluster_manager -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/ent/cluster_manager
Building wallaroo/core/metrics -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/core/metrics
Building wallaroo_labs/hub -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo_labs/hub
Building json -> /usr/lib/pony/0.21.0-4301.acd811b/packages/json
Building format -> /usr/lib/pony/0.21.0-4301.acd811b/packages/format
Building wallaroo/ent/router_registry -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/ent/router_registry
Building wallaroo/core/source -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/core/source
Building wallaroo/core/sink/tcp_sink -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/core/sink/tcp_sink
Building options -> /usr/lib/pony/0.21.0-4301.acd811b/packages/options
Building wallaroo/core/sink -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/core/sink
Building wallaroo/core/source/tcp_source -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/core/source/tcp_source
Building wallaroo_labs/dag -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo_labs/dag
Building wallaroo_labs/equality -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo_labs/equality
Building wallaroo_labs/messages -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo_labs/messages
Building wallaroo_labs/queue -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo_labs/queue
Building debug -> /usr/lib/pony/0.21.0-4301.acd811b/packages/debug
Building wallaroo_labs/thread_count -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo_labs/thread_count
Building wallaroo/core/state -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/core/state
Building wallaroo_labs/weighted -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo_labs/weighted
Building wallaroo_labs/collection_helpers -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo_labs/collection_helpers
Building wallaroo_labs/conversions -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo_labs/conversions
Building itertools -> /usr/lib/pony/0.21.0-4301.acd811b/packages/itertools
Building net/http -> /usr/lib/pony/0.21.0-4301.acd811b/packages/net/http
Building net/ssl -> /usr/lib/pony/0.21.0-4301.acd811b/packages/net/ssl
Building encode/base64 -> /usr/lib/pony/0.21.0-4301.acd811b/packages/encode/base64
Building pony-kafka -> /home/ubuntu/wallaroo-tutorial/wallaroo/machida/.deps/WallarooLabs/pony-kafka/pony-kafka
Building custombuffered -> /home/ubuntu/wallaroo-tutorial/wallaroo/machida/.deps/WallarooLabs/pony-kafka/pony-kafka/custombuffered
Building codecs -> /home/ubuntu/wallaroo-tutorial/wallaroo/machida/.deps/WallarooLabs/pony-kafka/pony-kafka/custombuffered/codecs
Building ../../utils/bool_converter -> /home/ubuntu/wallaroo-tutorial/wallaroo/machida/.deps/WallarooLabs/pony-kafka/pony-kafka/utils/bool_converter
Building customnet -> /home/ubuntu/wallaroo-tutorial/wallaroo/machida/.deps/WallarooLabs/pony-kafka/pony-kafka/customnet
Building customlogger -> /home/ubuntu/wallaroo-tutorial/wallaroo/machida/.deps/WallarooLabs/pony-kafka/pony-kafka/customlogger
Building fsm -> /home/ubuntu/wallaroo-tutorial/wallaroo/machida/.deps/WallarooLabs/pony-kafka/pony-kafka/fsm
Building compression -> /home/ubuntu/wallaroo-tutorial/wallaroo/machida/.deps/WallarooLabs/pony-kafka/pony-kafka/compression
Building wallaroo/core/sink/kafka_sink -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/core/sink/kafka_sink
Building wallaroo/core/source/kafka_source -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo/core/source/kafka_source
Generating
 Reachability
 Selector painting
 Data prototypes
 Data types
 Function prototypes
 Functions
 Descriptors
Optimising
Stack dump:
0.      Running pass 'Function Pass Manager' on module 'machida'.
1.      Running pass 'Loop Vectorization' on function '@Array_t2_U128_val_U128_val_Serialise'
Segmentation fault (core dumped)
../machida/Makefile:68: recipe for target '/home/ubuntu/wallaroo-tutorial/wallaroo/machida/build/machida' failed
make: *** [/home/ubuntu/wallaroo-tutorial/wallaroo/machida/build/machida] Error 1
```

hi @SeanTAllen 
Please find the terminal output below
```
ubuntu@ip-:~/wallaroo-tutorial/wallaroo/machida$ make
ar: creating /home/ubuntu/wallaroo-tutorial/wallaroo/machida/build/libpython-wal                                                                                                                laroo.a
a - /home/ubuntu/wallaroo-tutorial/wallaroo/machida/build/python-wallaroo.o
Cloning into '/home/ubuntu/wallaroo-tutorial/wallaroo/machida/.deps/WallarooLabs                                                                                                                /pony-kafka'...
remote: Counting objects: 372, done.
remote: Total 372 (delta 0), reused 0 (delta 0), pack-reused 372
Receiving objects: 100% (372/372), 443.11 KiB | 0 bytes/s, done.
Resolving deltas: 100% (235/235), done.
Checking connectivity... done.
Note: checking out '0.3.0'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at f738312... Compatiblity with ponyc master and exhaustive match ch                                                                                                                anges.
Building builtin -> /usr/lib/pony/0.21.0-4301.acd811b/packages/builtin
Building . -> /home/ubuntu/wallaroo-tutorial/wallaroo/machida
Building signals -> /usr/lib/pony/0.21.0-4301.acd811b/packages/signals
Building ponytest -> /usr/lib/pony/0.21.0-4301.acd811b/packages/ponytest
Building time -> /usr/lib/pony/0.21.0-4301.acd811b/packages/time
Building collections -> /usr/lib/pony/0.21.0-4301.acd811b/packages/collections
Building wallaroo_labs/mort -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/walla                                                                                                                roo_labs/mort
Building wallaroo_labs/options -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wa                                                                                                                llaroo_labs/options
Building wallaroo -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallaroo
Building wallaroo/ent/spike -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/walla                                                                                                                roo/ent/spike
Building wallaroo/ent/network -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wal                                                                                                                laroo/ent/network
Building wallaroo/core/common -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wal                                                                                                                laroo/core/common
Building wallaroo/core/boundary -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/w                                                                                                                allaroo/core/boundary
Building buffered -> /usr/lib/pony/0.21.0-4301.acd811b/packages/buffered
Building net -> /usr/lib/pony/0.21.0-4301.acd811b/packages/net
Building wallaroo_labs/bytes -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wall                                                                                                                aroo_labs/bytes
Building promises -> /usr/lib/pony/0.21.0-4301.acd811b/packages/promises
Building wallaroo_labs/time -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/walla                                                                                                                roo_labs/time
Building wallaroo/ent/watermarking -> /home/ubuntu/wallaroo-tutorial/wallaroo/li                                                                                                                b/wallaroo/ent/watermarking
Building wallaroo/core/invariant -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/                                                                                                                wallaroo/core/invariant
Building wallaroo/core/routing -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wa                                                                                                                llaroo/core/routing
Building wallaroo_labs/guid -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/walla                                                                                                                roo_labs/guid
Building random -> /usr/lib/pony/0.21.0-4301.acd811b/packages/random
Building wallaroo/core/messages -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/w                                                                                                                allaroo/core/messages
Building serialise -> /usr/lib/pony/0.21.0-4301.acd811b/packages/serialise
Building wallaroo/ent/data_receiver -> /home/ubuntu/wallaroo-tutorial/wallaroo/l                                                                                                                ib/wallaroo/ent/data_receiver
Building wallaroo/core/data_channel -> /home/ubuntu/wallaroo-tutorial/wallaroo/l                                                                                                                ib/wallaroo/core/data_channel
Building files -> /usr/lib/pony/0.21.0-4301.acd811b/packages/files
Building capsicum -> /usr/lib/pony/0.21.0-4301.acd811b/packages/capsicum
Building wallaroo/ent/recovery -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wa                                                                                                                llaroo/ent/recovery
Building wallaroo/core/topology -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/w                                                                                                                allaroo/core/topology
Building assert -> /usr/lib/pony/0.21.0-4301.acd811b/packages/assert
Building wallaroo/ent/rebalancing -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib                                                                                                                /wallaroo/ent/rebalancing
Building wallaroo/core/initialization -> /home/ubuntu/wallaroo-tutorial/wallaroo                                                                                                                /lib/wallaroo/core/initialization
Building wallaroo/ent/cluster_manager -> /home/ubuntu/wallaroo-tutorial/wallaroo                                                                                                                /lib/wallaroo/ent/cluster_manager
Building wallaroo/core/metrics -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wa                                                                                                                llaroo/core/metrics
Building wallaroo_labs/hub -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallar                                                                                                                oo_labs/hub
Building json -> /usr/lib/pony/0.21.0-4301.acd811b/packages/json
Building format -> /usr/lib/pony/0.21.0-4301.acd811b/packages/format
Building wallaroo/ent/router_registry -> /home/ubuntu/wallaroo-tutorial/wallaroo                                                                                                                /lib/wallaroo/ent/router_registry
Building wallaroo/core/source -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wal                                                                                                                laroo/core/source
Building wallaroo/core/sink/tcp_sink -> /home/ubuntu/wallaroo-tutorial/wallaroo/                                                                                                                lib/wallaroo/core/sink/tcp_sink
Building options -> /usr/lib/pony/0.21.0-4301.acd811b/packages/options
Building wallaroo/core/sink -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/walla                                                                                                                roo/core/sink
Building wallaroo/core/source/tcp_source -> /home/ubuntu/wallaroo-tutorial/walla                                                                                                                roo/lib/wallaroo/core/source/tcp_source
Building wallaroo_labs/dag -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wallar                                                                                                                oo_labs/dag
Building wallaroo_labs/equality -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/w                                                                                                                allaroo_labs/equality
Building wallaroo_labs/messages -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/w                                                                                                                allaroo_labs/messages
Building wallaroo_labs/queue -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wall                                                                                                                aroo_labs/queue
Building debug -> /usr/lib/pony/0.21.0-4301.acd811b/packages/debug
Building wallaroo_labs/thread_count -> /home/ubuntu/wallaroo-tutorial/wallaroo/l                                                                                                                ib/wallaroo_labs/thread_count
Building wallaroo/core/state -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/wall                                                                                                                aroo/core/state
Building wallaroo_labs/weighted -> /home/ubuntu/wallaroo-tutorial/wallaroo/lib/w                                                                                                                allaroo_labs/weighted
Building wallaroo_labs/collection_helpers -> /home/ubuntu/wallaroo-tutorial/wall                                                                                                                aroo/lib/wallaroo_labs/collection_helpers
Building wallaroo_labs/conversions -> /home/ubuntu/wallaroo-tutorial/wallaroo/li                                                                                                                b/wallaroo_labs/conversions
Building itertools -> /usr/lib/pony/0.21.0-4301.acd811b/packages/itertools
Building net/http -> /usr/lib/pony/0.21.0-4301.acd811b/packages/net/http
Building net/ssl -> /usr/lib/pony/0.21.0-4301.acd811b/packages/net/ssl
Building encode/base64 -> /usr/lib/pony/0.21.0-4301.acd811b/packages/encode/base                                                                                                                64
Building pony-kafka -> /home/ubuntu/wallaroo-tutorial/wallaroo/machida/.deps/Wal                                                                                                                larooLabs/pony-kafka/pony-kafka
Building custombuffered -> /home/ubuntu/wallaroo-tutorial/wallaroo/machida/.deps                                                                                                                /WallarooLabs/pony-kafka/pony-kafka/custombuffered
Building codecs -> /home/ubuntu/wallaroo-tutorial/wallaroo/machida/.deps/Wallaro                                                                                                                oLabs/pony-kafka/pony-kafka/custombuffered/codecs
Building ../../utils/bool_converter -> /home/ubuntu/wallaroo-tutorial/wallaroo/m                                                                                                                achida/.deps/WallarooLabs/pony-kafka/pony-kafka/utils/bool_converter
Building customnet -> /home/ubuntu/wallaroo-tutorial/wallaroo/machida/.deps/Wall                                                                                                                arooLabs/pony-kafka/pony-kafka/customnet
Building customlogger -> /home/ubuntu/wallaroo-tutorial/wallaroo/machida/.deps/W                                                                                                                allarooLabs/pony-kafka/pony-kafka/customlogger
Building fsm -> /home/ubuntu/wallaroo-tutorial/wallaroo/machida/.deps/WallarooLa                                                                                                                bs/pony-kafka/pony-kafka/fsm
Building compression -> /home/ubuntu/wallaroo-tutorial/wallaroo/machida/.deps/Wa                                                                                                                llarooLabs/pony-kafka/pony-kafka/compression
Building wallaroo/core/sink/kafka_sink -> /home/ubuntu/wallaroo-tutorial/wallaro                                                                                                                o/lib/wallaroo/core/sink/kafka_sink
Building wallaroo/core/source/kafka_source -> /home/ubuntu/wallaroo-tutorial/wal                                                                                                                laroo/lib/wallaroo/core/source/kafka_source
Generating
 Reachability
 Selector painting
 Data prototypes
 Data types
 Function prototypes
 Functions
 Descriptors
Optimising
Stack dump:
0.      Running pass 'Function Pass Manager' on module 'machida'.
1.      Running pass 'Loop Vectorization' on function '@Array_t2_U128_val_U128_val_Serialise'
Segmentation fault (core dumped)
../machida/Makefile:68: recipe for target '/home/ubuntu/wallaroo-tutorial/wallaroo/machida/build/machida' failed
make: *** [/home/ubuntu/wallaroo-tutorial/wallaroo/machida/build/machida] Error 1

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,8 @@ All notable changes to Wallaroo will be documented in this file.
 
 ### Changed
 
+- Turn off building with AVX512f CPU extensions to work around a LLVM bug ([PR #1932](https://github.com/WallarooLabs/wallaroo/pull/1932))
+
 ## [0.3.2] - 2017-12-28
 
 ### Fixed

--- a/rules.mk
+++ b/rules.mk
@@ -255,11 +255,11 @@ define PONYC
     $(if $(filter $(ponyc_docker_args),docker),$(quote))
   $(QUIET)cd $(1) && $(ponyc_docker_args) $(PONYSTABLE) env $(PONYCC) $(ponyc_arch_args) \
     $(debug_arg) $(spike_arg) $(autoscale_arg) $(clustering_arg) $(resilience_arg) \
-    $(PONYCFLAGS) $(target_cpu_arg) . $(if $(filter $(ponyc_docker_args),docker),$(quote))
+    $(PONYCFLAGS) $(target_cpu_arg) --features=-avx512f . $(if $(filter $(ponyc_docker_args),docker),$(quote))
   $(QUIET)cd $(1) && echo "$@: $(abspath $(1))/bundle.json" | tr '\n' ' ' > $(notdir $(abspath $(1:%/=%))).d
   $(QUIET)cd $(1) && $(ponyc_docker_args) $(PONYSTABLE) env $(PONYCC) $(ponyc_arch_args) \
     $(debug_arg) $(spike_arg) $(autoscale_arg) $(clustering_arg) $(resilience_arg) \
-    $(PONYCFLAGS) $(target_cpu_arg) . --pass import --files $(if $(filter \
+    $(PONYCFLAGS) $(target_cpu_arg) --features=-avx512f . --pass import --files $(if $(filter \
     $(ponyc_docker_args),docker),$(quote)) 2>/dev/null | grep -o "$(abs_wallaroo_dir).*.pony" \
     | awk 'BEGIN { a="" } {a=a$$1":\n"; printf "%s ",$$1} END {print "\n"a}' \
     >> $(notdir $(abspath $(1:%/=%))).d

